pkg/client-sdk/base_client.go

```diff
@@ -186,11 +197,28 @@ func (a *arkClient) NotifyIncomingFunds(
 		return nil, fmt.Errorf("wallet not initialized")
 	}
 
-	eventCh, closeFn, err := a.client.SubscribeForAddress(ctx, addr)
+	decoded, err := common.DecodeAddress(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	scripts := []string{
+		hex.EncodeToString(schnorr.SerializePubKey(decoded.VtxoTapKey)),
+	}
+	subId, err := a.indexer.SubscribeForScripts(ctx, "", scripts)
+	if err != nil {
+		return nil, err
+	}
+
+	eventCh, closeFn, err := a.indexer.GetSubscription(ctx, subId)
 	if err != nil {
 		return nil, err
 	}
-	defer closeFn()
+	defer func() {
+		// nolint
+		a.indexer.UnsubscribeForScripts(ctx, subId, scripts)
```

Comment:
General comment for ignoring errors:
i would rather log the error and optionally send it to sentry